### PR TITLE
Make Lago pip installable

### PR DESCRIPTION
**Pull request by nvgoldin**
Notable changes
1. Fixes of various requirements.txt issues.
2. Drop python-magic dependency
3. Change the SDK functional tests to consume the sdist generated by tox(with python setup.py sdist), so if that passes - it proves we are pip installable.
4. No more '--with-system-packages' will be needed in venv.

### CODE REVIEW

**Comment by nvgoldin** [comment]
The updated install procedure(pip section) can be viewed in:
https://github.com/nvgoldin/lago/blob/9644f5daa2771927d4c975b4fee44735c1a2b161/docs/Installation.rst

File: MANIFEST.in [hunk 1]
```diff
@@ -1 +1,11 @@
-include AUTHORS COPYING README.md
```
Comment (gbenhaim) [inline MANIFEST.in]: Why not using setup.cfg for this task ?

File: automation/check-patch.packages.el7 [hunk 2]
```diff
@@ -2,13 +2,13 @@ ansible
...
 git
 libffi-devel
+libguestfs-devel
 libvirt-devel
 libxslt-devel
 openssl-devel
 pandoc
 python2-devel
 python-dulwich
-python-libguestfs
```
Comment (gbenhaim) [inline automation/check-patch.packages.el7]: Why is this change needed?

File: docs/Installation.rst [hunk 3]
```diff
@@ -1,90 +1,224 @@
...
+
+   .. code:: bash
+
+       $ yum install -y libvirt libvirt-devel libguestfs-tools \
+           libguestfs-devel gcc libffi-devel openssl-devel qemu-kvm
+
+   C. Debian / Ubuntu - *TO-DO*
+   D. ArchLinux - *TO-DO*
+
+2. Install libguestfs python bindings, as they are not available on PyPI [3]_:
```
Comment (gbenhaim) [inline docs/Installation.rst]: I think that we should install this library as part of the system packages.
For Debian we can use python-guestfs

File: setup.py [hunk 4]
```diff
@@ -58,6 +58,6 @@ def get_version(project_dir=os.curdir):
 if __name__ == '__main__':
     os.environ['PBR_VERSION'] = get_version()
     setup(
-        setup_requires=['pbr'],
+        setup_requires=['pbr', 'dulwich'],
```
Comment (gbenhaim) [inline setup.py]: Why is this needed? version_manager.py ?

File: MANIFEST.in (hunk 1 above)
Comment (nvgoldin) [inline MANIFEST.in]: Right missed that, according to http://setuptools.readthedocs.io/en/latest/setuptools.html#including-data-files I can use 'exclude' directive. Will change.

File: automation/check-patch.packages.el7 (hunk 2 above)
Comment (nvgoldin) [inline automation/check-patch.packages.el7]: Just to  be careful - I didn't want python-libguestfs to be available on the system for the unit testing. When we'll run the functional CLI tests, we install from RPM anyways so it will find its way to there.

File: docs/Installation.rst (hunk 3 above)
Comment (nvgoldin) [inline docs/Installation.rst]: The disadvantages of a system package:
1. Virtualenv - you need to instruct users to use --with-system-packages, which breaks the concept of a virtualenv. 
2. We tie ourselves to a distro specific package, while it might not be available on every distro(though I agree in this case - if someone built 'libguestfs' package which we need anyways, he would probably build the python binding also, but that is not promised).
This is our only Python system package that we require. Without it you can really install Lago in a virtualenv, without any hacks.

In my opinion,  for pip instructions - we should strive to install whatever is Python in the Python way of things - pip, virtualenv, tox, etc. 

With all that being said, it really doesn't matter - if someone wants to use the system-package he can always do it. I can add it to the notes.

File: setup.py (hunk 4 above)
Comment (nvgoldin) [inline setup.py]: Yep.
Actually its weird - I think ``pbr`` should pull it. I'll give it another try.

File: MANIFEST.in (hunk 1 above)
Comment (nvgoldin) [inline MANIFEST.in]: Seems like ``pbr`` doesn't support it - according to the docs it will attempt to auto-generate the list, filtering .gitignore files, but it doesn't allow custom filtering
https://docs.openstack.org/developer/pbr/#manifest

So we probably still need `MANIFEST.in`.

File: setup.py (hunk 4 above)
Comment (nvgoldin) [inline setup.py]: OK, so regardless if `pbr` pulls it or not, we have an explicit import in the version_manager.py, so I'll keep it.

**Comment by nvgoldin** [comment]
@gbenhaim  - GitHub erased my reply to your comment about libguestfs on the rebase, so here it is:
>docs/Installation.rst
>+   C. Debian / Ubuntu - *TO-DO*
>+   D. ArchLinux - *TO-DO*
>+
>+2. Install libguestfs python bindings, as they are not available on PyPI [3]_:
>@gbenhaim
>gbenhaim 20 hours ago Owner
>
>I think that we should install this library as part of the system packages.
>For Debian we can use python-guestfs
>@nvgoldin
>nvgoldin 13 hours ago • edited Owner
>
>The disadvantages of a system package:
>
>    Virtualenv - you need to instruct users to use --with-system-packages, which breaks the concept of a >virtualenv.
>    We tie ourselves to a distro specific package, while it might not be available on every distro(though I agree in this case - if someone built 'libguestfs' package which we need anyways, he would probably build the python binding also, but that is not promised).
>This is our only Python system package that we require. Without it you can really install Lago in a virtualenv, without any hacks.
>In my opinion, for pip instructions - we should strive to install whatever is Python in the Python way of things - pip, virtualenv, tox, etc.
>With all that being said, it really doesn't matter - if someone wants to use the system-package he can always do it. I can add it to the notes.


Update for this version - I updated the docs to clarify that point.

**Comment by nvgoldin** [comment]
ci merge please

**Comment by nvgoldin** [comment]
With regard to https://github.com/lago-project/lago/pull/570#issuecomment-309365061 we discussed it offline and agreed this is only a docs concern - so people could still use the system package if they want. Also agreed we'll implement https://github.com/lago-project/lago/issues/573 and make guestfs optional.

**Comment by nvgoldin** [comment]
ci merge please
